src/engine/symbolic_execution.rs

```diff
@@ -778,28 +983,24 @@ where
             (Value::Symbolic(v1), Value::Symbolic(v2)) => {
                 self.execute_beq_branches(true_branch, false_branch, v1, v2)
             }
-            (v1, v2) => {
-                self.is_running = false;
-
-                let result = self.check_for_uninitialized_memory(Instruction::Beq(btype), v1, v2);
-
-                trace!("access to uninitialized memory => exiting this context");
-
-                result
-            }
+            (v1, v2) => self.access_to_uninitialized_memory(Instruction::Beq(btype), v1, v2),
         }
     }
 
-    fn execute_exit(&mut self) -> Result<Option<Bug>, SymbolicExecutionError> {
-        self.is_running = false;
-
+    fn execute_exit(&mut self) -> Result<(), SymbolicExecutionError> {
         match self.regs[Register::A0 as usize] {
             Value::Symbolic(exit_code) => {
                 trace!("exit: symbolic code -> find input for exit_code != 0");
 
-                self.execute_query(Query::NotEquals((exit_code, 0)), |info| {
-                    Bug::ExitCodeGreaterZero { info }
-                })
+                self.execute_query(Query::NotEquals((exit_code, 0)))
+                    .and_then(|r| {
+                        self.exit_anyway_with_bug_if_sat(
+                            r,
+                            // TODO: return symbolic exit code in bug
+                            |i| self.exit_code_greater_zero_bug(i, 0),
```

Comment:
nit: Lets at least use a non-zero exit code in the Sat case. This was throwing me off despite the TODO, just `1` or `-1` is totally fine.

Reply:
Yes! I think, I'll have a look at this problem, before we merge that. Any hard coded value is wrong and will lead to confusion.

Reply:
actually returning concrete/symbolic values in a bug now.

Reply:
Nice. Thanks!